src/edit/main.js

```diff
@@ -156,11 +156,22 @@ class ProseMirror {
       domDrop: new DOMSubscription
     }
 
-    if (opts.place && opts.place.appendChild)
+    if (opts.place && opts.place.appendChild) {
+      this.root = opts.place
+
+      // loop if root is not document (in light dom)
+      // only shadowRoot has property (in shadow dom)
+      while (this.root!== document && !this.root.host) {
```

Comment:
It looks like all this code should be moved to after the if/else block so that the case when `opts.place` is a function is also catered for.